src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -592,12 +610,27 @@ private void addItem(
     RoutingContext routingContext,
     WebContext webContext,
     Item newItem,
-    ItemCollection itemCollection) {
+    ItemCollection itemCollection,
+    UserCollection userCollection) {
+
+    String userId = routingContext.request().getHeader("X-Okapi-User-Id");
+    userCollection.findById(userId,
+      userSuccess -> {
+        User user = userSuccess.getResult();
+
+        List<CirculationNote> notes = newItem.getCirculationNotes()
+          .stream()
+          .map(note -> note.withId(UUID.randomUUID().toString()))
+          .map(note -> note.withSource(user))
+          .map(note -> note.withDate(dateTimeFormatter.format(ZonedDateTime.now())))
+          .collect(Collectors.toList());
 
-    itemCollection.add(newItem, success -> {
-      Item item = success.getResult();
-      respondWithItemRepresentation(item, STATUS_CREATED, routingContext, webContext);
-    }, FailureResponseConsumer.serverError(routingContext.response()));
+        itemCollection.add(newItem.withCirculationNotes(notes), success -> {
+          Item item = success.getResult();
+          respondWithItemRepresentation(item, STATUS_CREATED, routingContext, webContext);
+        }, FailureResponseConsumer.serverError(routingContext.response()));
+      },
+      FailureResponseConsumer.serverError(routingContext.response()));
```

Comment:
Does this mean that if the user cannot be found, the whole request to create the instance will fail?

Reply:
Yes, it does.

Reply:
This means that instances cannot be created without a user header (and is dependent upon the availability of the module providing the `users` interface)

Does the data import create circulation notes? If so, does it run in the context of a user (I imagine it is a background process)?

Reply:
What do you mean by data import?